src/Sylius/Bundle/CoreBundle/Sitemap/Model/SitemapUrl.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+ 
+namespace Sylius\Bundle\CoreBundle\Sitemap\Model;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class SitemapUrl implements SitemapUrlInterface
+{
+    /**
+     * @var string
+     */
+    private $localization;
+
+    /**
+     * @var \DateTime
+     */
+    private $lastModification;
+
+    /**
+     * @var string
+     */
+    private $changeFrequency;
+
+    /**
+     * @var float
+     */
+    private $priority;
+
+    /**
+     * @var string
+     */
+    private $template;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLocalization()
+    {
+        return $this->localization;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setLocalization($localization)
+    {
+        $this->localization = $localization;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLastModification()
+    {
+        return $this->lastModification;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setLastModification(\DateTime $lastModification)
+    {
+        $this->lastModification = $lastModification;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getChangeFrequency()
+    {
+        return $this->changeFrequency;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setChangeFrequency($changeFrequency)
+    {
+        if (!in_array($changeFrequency, self::getSupportedChangeFrequencies(), true)) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'The value %s is not supported by the option changefreq.',
+                    $changeFrequency
+                )
+            );
+        }
+
+        $this->changeFrequency = $changeFrequency;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        return $this->priority;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setPriority($priority)
+    {
+        if (!is_numeric($priority) || 0 >= $priority || 1 <= $priority) {
+            throw new \InvalidArgumentException(
+                sprintf(
```

Comment:
I think, there is no point in spliting this exception to so many lines. I would suggest to write it in one line.